lib/linter/linter.js

```diff
@@ -269,12 +276,22 @@ function getDirectiveComments(filename, ast, ruleMapper) {
 
     ast.comments.filter(token => token.type !== "Shebang").forEach(comment => {
         const trimmedCommentText = comment.value.trim();
-        const match = /^(eslint(-\w+){0,3}|exported|globals?)(\s|$)/u.exec(trimmedCommentText);
+        const match = /^(eslint(?:-env|-enable|-disable(?:(?:-next)?-line)?)?|exported|globals?)(?:\s|$)/u.exec(trimmedCommentText);
 
         if (!match) {
             return;
         }
 
+        if (warnInlineConfig) {
+            problems.push(createLintingProblem({
+                ruleId: null,
+                message: `'/*${match[1]}*/' has no effects because you have 'noInlineConfig' setting in your config.`,
```

Comment:
Small typo:

```suggestion
                message: `'/*${match[1]}*/' has no effect because you have 'noInlineConfig' setting in your config.`,
```

More generally, I'm worried about whether we should include which config file(s) enabled the setting in the error message (but I know that would be a huge pain to find in Linter).